### PR TITLE
Delete project usecase implementation

### PR DESCRIPTION
**Pull request by bing-j**
Code associated with #5. Note that no tests have been written yet.

### CODE REVIEW

**Review by skrtsamjang — COMMENTED** [review]
I think it's all good, I was wondering at first how you would handle if there were no projects to delete, but I guess you are handling the deletion inside the projectview and not the dashboard view, then which you won't need a fail case of deleting projects. You might have to check with Vedant to see if the layout of the project view is matching your use case.

File: src/use_case/delete_project/DeleteProjectDataAccessInterface.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package use_case.delete_project;
+
+import entity.User;
+
+public interface DeleteProjectDataAccessInterface {
+
+    User getCurrentUser();
```
Comment (BraedenChoo) [inline src/use_case/delete_project/DeleteProjectDataAccessInterface.java:7]: I'm sure you know but just so you remember, this method is deprecated.

File: src/use_case/delete_project/DeleteProjectInteractor.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    public DeleteProjectInteractor(DeleteProjectDataAccessInterface userDataAccessObject, DeleteProjectOutputBoundary deleteProjectPresenter) {
+        this.userDataAccessObject = userDataAccessObject;
+        this.deleteProjectPresenter = deleteProjectPresenter;
+    }
+
+    @Override
+    public void execute(DeleteProjectInputData deleteProjectInputData) {
+        String projectId = deleteProjectInputData.getProjectId();
+        User currentUser = userDataAccessObject.getCurrentUser();
```
Comment (BraedenChoo) [inline src/use_case/delete_project/DeleteProjectInteractor.java:20]: just a reminder to refactor this for getUser(String)

**Review by BraedenChoo — COMMENTED** [review]
Nice work, but there are a few formatting and syntax errors I've pointed out you should resolve first.

**Review by skrtsamjang — CHANGES_REQUESTED** [review]
I think it's pretty good, just some minor convention mistakes as well as some redundant methods, but other than that great job.

**Review by BraedenChoo — APPROVED** [review]
Looks good now!

**Review by skrtsamjang — APPROVED** [review]
i believe it works now

**Pull request merged**